src/librustc_hir/def.rs

```diff
@@ -88,6 +88,7 @@ pub enum DefKind {
     GlobalAsm,
     Impl,
     Closure,
+    Crate,
```

Comment:
Ok, this is something I disagree with.

Paths *do* resolve to the crate root already (`crate` or `$crate`) and they use `DefKind::Mod` for that.

Reply:
Oh, got it. So I should return `DefKind::Mod` instead? Or just not resolve the `def_kind` of a crate root?

Reply:
The former, everything should have a `def_kind`.

Reply:
From what I see, nothing prevents calling `fn def_kind` with a crate root `LocalDefId`.
E.g. it could easily happen through some `def_kind(id).descr()` in diagnostics.
If it doesn't actually happen right now, then it's rather a happy accident than a rule.
So, it should be reasonable to return `DefKind::Mod`.